docs/develop.md

```diff
@@ -1,18 +1,426 @@
+to as the "remote" copy. Navigate to (your fork of) the WAM2layers repo and find
+the big green button that says "code". Now, there are a few options.
+
+Cloning with HTTPS works out of the box. However, whenever you want to "push"
+changes back to the remote, you'll need to authenticate. This can be annoying.
+
+Cloning with SSH is very convenient, but you have to [set up an SSH key
+first](https://docs.github.com/en/authentication/connecting-to-github-with-ssh).
+If you work on different machines, you can set multiple SSH keys. Despite the
+initial efforts, in our experience this is the preferred way to authenticate
+with GitHub.
+
+There is also the option to use the GitHub command line interface. This is a
+[separate software](https://cli.github.com/) provided by GitHub. We have little
+experience with it.
+
+Once you've decided how to authenticate, copy the corresponding url. Then open a
+terminal and paste the url preceded with `git clone`. In case of using SSH:
+
+```
+# With fork
+git clone git@github.com:Peter9192/WAM2layers.git
+
+# Without fork
+git clone git@github.com:WAM2layers/WAM2layers.git
+```
+
+### Development installation - additional tooling
+
+You can install your local copy of the code by following the same steps as the
+[regular installation](./installation.md). However, by adding `[develop]`, some
+additional dependencies are installed that are convenient for developing.
+
+```
+pip install --editable .[develop]
+```
+
+The `--editable` flag means that any changes you make to the code will be
+"effective immediately". When you run the model again, the updated code is used.
+
+The `[develop]` option tells pip to install not just the code, but some
+additional packages that are listed under the "develop" header in the file
+`pyproject.toml`. These packages help with linting (checking your code against
+syntax/style guides), automatic formatting, building documentation, running
+tests, and publising the package on PyPI.
+
+### Create a new branch
+
+It is good practice to create a new "branch" for each feature you are adding.
+For example, this will create (and switch to) a new branch called
+"developer-documentation":
+
+```
+git checkout -b developer-documentation
+```
+
+You can use `git branch` to see all your branches, and
+`git checkout <branchname>` (without the `-b`) to switch to another (existing)
+branch. In case of the example, I would use this branch while working on this
+documentation page. But if I was interrupting my work to fix a bug on another
+part of the code, I would switch to another branch. This keeps all changes that
+belong together, well, together.
+
+### Make changes
+
+Now you are ready to make the changes you need and test whether they work. Keep
+in mind that your changes should not break the code for other people. If you're
+adding new code, please also add relevant documentation, and add an update to
+the changelog. If your introducing new dependencies, add them to
```

Comment:
Great to mention and remind us of the changelog. Would be good to then to follow this example.

Reply:
Haha, nice catch